kedro/framework/cli/catalog.py

```diff
@@ -174,3 +176,64 @@ def _add_missing_datasets_to_catalog(missing_ds, catalog_path):
     catalog_path.parent.mkdir(exist_ok=True)
     with catalog_path.open(mode="w") as catalog_file:
         yaml.safe_dump(catalog_config, catalog_file, default_flow_style=False)
+
+
+def pick_best_match(matches):
+    matches = sorted(matches, key=lambda x: (specificity(x[0]), -x[0].count("{"), x[0]))
+    return matches[0]
+
+
+def specificity(pattern):
```

Comment:
Is this method just counting the number of characters that are not in brackets? Wouldn't it be easier to just remove the text in {} and then just get the length of the string?

Reply:
That is what is happening here but using `parse` instead of a regex.